### DISCUSSION TITLE
Problem - the tensor should have ### values but has 0

### DISCUSSION THREAD

**Discussion by TheRankin** [vladmandic/human#225] [Q&A] [answered]
I literally copy/paste the code from the "demo/typescript" and removed unnecessary code (those to show information as fps, etc) and am getting

-NodeJS - Ok
-Webcam - Ok
-Models - Ok

```ts
Uncaught (in promise) Error: Based on the provided shape, [1,1,96,2], the tensor should have 192 values but has 0
    at assert (human.esm.js:5709)
    at makeTensor (human.esm.js:8289)
    at tensor (human.esm.js:8305)
    at Object.decodeWeights (human.esm.js:8448)
    at GraphModel.loadSync (human.esm.js:37757)
    at GraphModel.load (human.esm.js:37744)
    at async loadGraphModel (human.esm.js:37853)
    at async load5 (human.esm.js:75436)
    at async load19 (human.esm.js:81414)
    at async Human.load (human.esm.js:83800)

Uncaught (in promise) Error: Based on the provided shape, [3,3,3,16], the tensor should have 432 values but has 0
    at assert (human.esm.js:5709)
    at makeTensor (human.esm.js:8289)
    at tensor (human.esm.js:8305)
    at Object.decodeWeights (human.esm.js:8448)
    at GraphModel.loadSync (human.esm.js:37757)
    at GraphModel.load (human.esm.js:37744)
    at async loadGraphModel (human.esm.js:37853)
    at async load11 (human.esm.js:76367)
```

**Comment by vladmandic** [comment]
this is strange - seems like loading of model weights can you
a) check browser inspector network view to see if weights got loaded correctly?
b) enable js map so backtrace shows actual code locations?
c) post log from browser inspector

**Reply by TheRankin** [reply]
well, I get a bug when I start the project, automatically the browser download the models (really weird)

a)
![image](https://user-images.githubusercontent.com/24502400/144955565-89e0c56f-6d46-4b72-b0ff-d8ae57b9f6d0.png)

b)
how?

c)
![image](https://user-images.githubusercontent.com/24502400/144955638-702ed31a-4614-4a9d-af82-a41c9c3ff42e.png)

**Comment by vladmandic** [comment]
check your dev web server - its not serving binary files (model weights) correctly - in network status it shows **status code 204** which is `No Content` (basically, your web server delivered an empty file instead of actual binary)

(unrelated, when posting logs, copy & paste text, not image)

**Reply by TheRankin** [reply]
this isn't correct? am using webpack and those models if from the models folder of this repo
![image](https://user-images.githubusercontent.com/24502400/144957882-7e6532be-15b0-4292-b9d1-d495ed70b9da.png)

**Comment by vladmandic** [comment]
paths seem correct since it loads model definitions correctly (`json` files) and it can find weights (`bin` files), but for some reason your `webpack` is NOT serving them correctly - it sends empty file as response. i don't have that issue (and plenty of other people are using `webpack` with `human` without issues) nor i'm about to troubleshoot webpack's internal web dev server - search on webpack forums whats typically wrong with it serving binary files.

**Reply by TheRankin** [reply]
well, when I refresh the page, the browser try to download the models, maybe is webpack :S

I was trying at the first time with Parcel but take a lot to bundle the project (never stop)

**Comment by vladmandic** [comment, marked as answer]
it definitely is the webpack configuration issue. i don't know why, but its not serving binary files correctly.
you can always use built in dev server that ships with `human` (my own, `@vladmandic/build`), you don't need either parcel or webpack.

**Reply by TheRankin** [reply]
ok ok, I'll try it

**Reply by TheRankin** [reply]
I tested on Parcel first (because I know how to use it fast), same situation but there is more information and is using the bins
![image](https://user-images.githubusercontent.com/24502400/144960735-d616281d-8602-4a4c-9a62-3a2a514ed4d2.png)

**Reply by TheRankin** [reply]
actually setting backend: cpu fix those backend errors, but the tensor problem exist yet

**Reply by vladmandic** [reply]
Your parcel is creating NodeJS package, not browser! That's why Human is trying wrong backend.
Manually setting backend to CPU is not a solution, you really need to fix your parcel altogether. It's messing everything.

**Reply by TheRankin** [reply]
Issue founded, I was running the project on a pwa-chrome so for some strange reason the models are auto-downloaded and the TensorFlow doesn't work properly

**Reply by TheRankin** [reply]
I accidentally opened the site in my normal browser and noticed that it worked

**Reply by vladmandic** [reply]
> for some strange reason the models are auto-downloaded

nothing strange about it  
models are auto-downloaded first time they are needed (and in demo case, it calls warmup function)  

> and the TensorFlow doesn't work properly

don't make assumptions like that  
nothing wrong with TensorFlow, issue is DOWNLOAD of binary files  
you can see code 204 and incorrect size

**Reply by TheRankin** [reply]
-No no, I mean auto-downloaded as a normal file, requesting a folder to save the file🤔
-By that, I meant that the error was said that the tensor was not receiving data properly😅

**Reply by vladmandic** [reply]
> -No no, I mean auto-downloaded as a normal file, requesting a folder to save the file🤔

Ahhhh...That's new - why don't you actually describe the problem and post all logs at the same time. This is quite important.
It explains code 204 since when you click cancel, result is rejected so it appears as "no content".
But why would your browser open save-as dialog on a totally normal `fetch` request? Perhaps wrong MIME type in webpack/parcel for .bin files?

> -By that, I meant that the error was said that the tensor was not receiving data properly😅

Because it never actually received data - that's not a fault of TensorFlow.
